src/storage/src/source/postgres.rs

```diff
@@ -405,7 +407,7 @@ async fn postgres_replication_loop_inner(
                     &task_info.source_id
                 );
             }
-            Err(ReplicationError::Recoverable(e)) => {
+            Err(ReplicationError::Indefinite(e)) => {
```

Comment:
i wouldn't describe this as `Indefinite`, as it never errors the source (same with the retrying in the replication stream below); what if we made the 2 variants `Retryable` and `Definite`?

Reply:
No objection! Curious: why does "indefinite" imply that it should error the source?

Reply:
yeah +1.  I like sticking with one set of semantics here: definite vs indefinite/nondefinite seems to be wording we use throughout the code?  Unless, of course, there's a reason missing that we shouldn't!